core/vm/instructions.go

```diff
@@ -440,10 +440,10 @@ func opBlockhash(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) (
 	}
 	var upper, lower uint64
 	upper = interpreter.evm.Context.BlockNumber.Uint64()
-	if upper < 257 {
+	if upper < 1 {
```

Comment:
This should be 2, right?

```suggestion
	if upper < 2 {
```

Reply:
Both versions result in `lower = 0`.

Reply:
Changed to `< 2`, because it's more consistent with previous logic